src/fmeval/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,67 @@
+import ray.data
+from typing import List, Union
+
+from fmeval.constants import TRANSFORM_PIPELINE_MAX_SIZE
+from fmeval.exceptions import EvalAlgorithmClientError
+from fmeval.transforms.transform import Transform
+from fmeval.util import require
+
+NestedTransform = Union[Transform, "TransformPipeline"]
+
+
+class TransformPipeline:
+    def __init__(self, nested_transforms: List[NestedTransform]):
+        require(
+            isinstance(nested_transforms, List),
+            "TransformPipeline initializer accepts a list containing Transforms or TransformPipelines.",
```

Comment:
nit: Adding the incorrect input type in the error message here could be helpful for debugging (similar to the `nested_transformers` type error message below).